SeattleColleges/nsc-events-nestjs
# NSC-events-nest.js_01-132-error-handling-methods

Related to this user story [132](https://github.com/SeattleColleges/nsc-events-android/issues/132)
Resolves issue #[156](https://github.com/SeattleColleges/nsc-events-android/issues/156)

Added exception handling to the these methods (createEvent, updateActivityById, and DeleteActivityById) returning the proper exceptions when incorrect or non exist fields are entered, and added messages to the returns informing event creators when an event has successfully been created, updated, deleted. 
Update activity services test file to reflect changes made in activity service modules.


To test: 
- First, ensure connection to backend is setup through this [link](https://nextjsdeployment.notion.site/Connection-f810cd7541064cdeab6b731a5da32893)
- Then run backend locally by running command npm run start:dev
- In Postman, make a POST call to this endpoint http://localhost:3000/api/auth/signup to create an account. The returned token should be inserted in as a bearer token in the Authorization or Headers tab when logging in...
- Using the returned token from the sign-up route, use this endpoint http://localhost:3000/api/auth/login to login. And use the provided token and inserted in as a bearer token in the Authorization or Headers tab for event related calls
- Then, after logging in, test these endpoints using that token:

    - Creating a new event: POST: http://localhost:3000/api/events/new 
    - <img width="1199" alt="NewEventCreatedWithMessage" src="https://github.com/SeattleColleges/nsc-events-nestjs/assets/116779961/751f34c3-1ced-4ede-8e3f-a8b63f10683a">
    - _example entry below:_ 
    - {
        "eventTitle": "second event created now",
        "eventDescription": "Second back to school event",
        "eventCategory": "School event",
        "eventDate": "2022-06-05T00:00:00.000Z",
        "eventStartTime": "5:00 PM",
        "eventEndTime": "7:00 PM",
        "eventLocation": "Campus",
        "eventCoverPhoto": "www.nscevents.com/photo",
        "eventHost": "NSC is the host",
        "eventWebsite": "www.nscevents.com",
        "eventRegistration": "register",
        "eventCapacity": "30",
        "eventCost": "$5",
        "eventTags": [
            "school"
        ],
        "eventSpeakers": [
            "SpeakerOne",
            "SpeakerTwo"
        ],
        "eventContact": "nsc@gmail.com",
        "eventSocialMedia": {
            "Facebook": "NSC",
            "Twitter": "NSC"
        },
        "eventAccessibility": "Everyone's welcome",
        "isHidden": false,
    }


    - Updating an existing event: PUT: http://localhost:3000/api/events/update/:id 
    - _removing a character in the id_ <img width="1452" alt="IncorrectUpdatedIDEntered" src="https://github.com/SeattleColleges/nsc-events-nestjs/assets/116779961/c93fd5e6-69d7-4a51-af4a-1b4fd6d78857">
    - <img width="1467" alt="EventUpdatedWithMessage" src="https://github.com/SeattleColleges/nsc-events-nestjs/assets/116779961/8f623cb5-8da9-4403-a7de-19d1910e131b">


    - Deleting an event: DELETE: http://localhost:3000/api/events/remove/:id
    - _removing a character in the id_ <img width="1681" alt="IncorretDeletedIDEntered" src="https://github.com/SeattleColleges/nsc-events-nestjs/assets/116779961/047ff3cb-3bd5-4d49-bafb-6d2613249474">
    - <img width="1690" alt="EventDeletedWithMessage" src="https://github.com/SeattleColleges/nsc-events-nestjs/assets/116779961/07a1c2a4-bb5f-4399-8aac-566d46c47545">
